test/unit/pool/test-change-user-eject.js

```diff
@@ -38,17 +44,20 @@ server.listen(common.fakeServerPort, function(err) {
 
   pool.getConnection(function(err, conn1) {
     assert.ifError(err);
-    assert.strictEqual(conn1.threadId, 3);
+    assert.ok(conn1.threadId === 1 || conn1.threadId === 2);
+    assert.strictEqual(conn1.config.user, 'user_1');
```

Comment:
The test needs to run the query `SELECT CURRENT_USER()` to be sure the server-side actually got the change.